x-pack/platform/plugins/shared/streams/server/routes/internal/streams/systems/route.ts

```diff
@@ -329,11 +332,79 @@ export const identifySystemsRoute = createServerRoute({
   },
 });
 
+export const describeStreamRoute = createServerRoute({
+  endpoint: 'POST /internal/streams/{name}/_describe_stream',
+  options: {
+    access: 'internal',
+    summary: 'Generate a stream description',
+    description: 'Generate a stream description based on data in the stream',
+  },
+  security: {
+    authz: {
+      requiredPrivileges: [STREAMS_API_PRIVILEGES.read],
+    },
+  },
+  params: z.object({
+    path: z.object({ name: z.string() }),
+    query: z.object({
+      connectorId: z.string(),
+      from: dateFromString,
+      to: dateFromString,
+    }),
+  }),
+  handler: async ({
+    params,
+    request,
+    getScopedClients,
+    server,
+  }): Promise<Observable<StreamDescriptionEvent>> => {
+    const { scopedClusterClient, licensing, uiSettingsClient, streamsClient, inferenceClient } =
+      await getScopedClients({
+        request,
+      });
+
+    await assertSignificantEventsAccess({ server, licensing, uiSettingsClient });
```

Comment:
Is this tightly coupled with sig events/system detection? Or is this better surfaced as a license check?

Reply:
> Is this tightly coupled with sig events/system detection? Or is this better surfaced as a license check?

for now, yes, just one bucket